src/mappings/handlers/pureProxyEventHandler.ts

```diff
@@ -1,31 +1,39 @@
 import { SubstrateEvent } from "@subql/types";
 
-import { PureProxy } from "../../types";
+import { Proxied, PureProxy } from "../../types";
+import { extractProxyEventData } from "../../utils/extractProxyEventData";
+import { u8aToHex } from "@polkadot/util";
+import { decodeAddress } from "../../utils";
 
 export async function handlePureProxyEvent(
   event: SubstrateEvent
 ): Promise<void> {
-  const {
-    event: {
-      data: [accountId],
-    },
-  } = event;
+  const proxyData = extractProxyEventData(event);
 
-  if (!accountId) return;
+  if (!proxyData) return;
+
+  const { proxyAccountId, accountId, type, delay, blockNumber, extrinsicIndex } = proxyData;
 
   const pureProxy = PureProxy.create({
-    id: accountId.toHex(),
-    blockNumber: blockNumber(event),
-    extrinsicIndex: extrinsicIndex(event),
+    id: accountId,
+    chainId,
+    blockNumber,
+    extrinsicIndex,
   });
 
   await pureProxy.save();
-}
 
-function extrinsicIndex(event: SubstrateEvent): number {
-  return event.extrinsic ? event.extrinsic.idx : event.idx;
-}
+  const proxy = Proxied.create({
+    id: `${chainId}-${accountId}-${proxyAccountId}-${type}-${delay}`,
+    chainId,
+    type,
+    proxyAccountId: u8aToHex(decodeAddress(proxyAccountId)),
+    accountId: u8aToHex(decodeAddress(accountId)),
```

Comment:
why do we use that conversion only here? I would suggesto to add it in extractProxyEventData, due to the fact that it also used in many places, where proxied entity is created